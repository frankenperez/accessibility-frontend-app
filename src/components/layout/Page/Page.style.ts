import styled from "@emotion/styled";

const StyledPage = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100vh;

  main {
    display: flex;
    flex-direction: row;
    flex-grow: 1;
    position: relative;
    background: $color__mono2;
    width: 100%;
    overflow: hidden;
  }
`;

export default StyledPage;
