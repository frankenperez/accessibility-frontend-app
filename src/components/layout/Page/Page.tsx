// React
import React from "react";

// Layout Components
import Header from "components/layout/Header/Header";
import Footer from "components/layout/Footer/Footer";

// Styles
import StyledPage from "./Page.style";

// Definitions
export interface PageProps {
  children?: React.ReactNode;
  showHeader?: boolean;
}

export default function Page({ children }: PageProps) {
  return (
    <StyledPage>
      <Header />
      <main>{children}</main>
      <Footer />
    </StyledPage>
  );
}
