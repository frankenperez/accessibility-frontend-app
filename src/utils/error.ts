import axios from "axios";

export const errorParser = (error: unknown) => {
  let message: string;
  if (axios.isAxiosError(error)) {
    const errorData = error.response?.data as { reason: string } | string;
    if (typeof errorData === "object" && errorData?.reason) {
      message = errorData?.reason;
    } else {
      message = JSON.stringify(errorData);
    }
  } else if (error instanceof Error) {
    message = error.message;
  } else {
    message = JSON.stringify(error);
  }
  return message;
};
