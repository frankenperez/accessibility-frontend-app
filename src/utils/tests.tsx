// React
import React, { PropsWithChildren } from "react";

// Testing Library
import { render, RenderOptions } from "@testing-library/react";

// Router
import { BrowserRouter } from "react-router-dom";

// Internationalization
import en_BG from "../languages/en-GB.json";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";

// MUI Theme
import { createTheme, StyledEngineProvider, ThemeProvider } from "@mui/material/styles";
import theme from "styles/theme";
const appTheme = createTheme(theme);

// Internationalization config
i18next.init({
  interpolation: { escapeValue: false }, // React already does escaping
  lng: "en", // Language to use
  resources: {
    en: {
      app: en_BG
    }
  }
});

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  route?: string;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => {
    return (
      <I18nextProvider i18n={i18next}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={appTheme}>
            <BrowserRouter>{children}</BrowserRouter>
          </ThemeProvider>
        </StyledEngineProvider>
      </I18nextProvider>
    );
  };
  window.history.pushState({}, "Test page", route);

  // Return an object with the store and all of RTL's query functions
  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
