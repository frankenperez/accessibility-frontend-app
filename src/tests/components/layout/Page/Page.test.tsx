// React
import React from "react";

// Component
import Page from "components/layout/Page/Page";

// Test Utils
import { renderWithProviders } from "utils/tests";

// Tests
describe("Tests for <Page />", () => {
  it("Renders with default props", () => {
    const { container } = renderWithProviders(<Page />);
    // Creates the snapshot test
    expect(container).toMatchSnapshot();
  });
});
