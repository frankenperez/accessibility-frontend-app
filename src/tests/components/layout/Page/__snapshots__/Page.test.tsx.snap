// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Page /> Renders with default props 1`] = `
<div>
  <div
    class="css-1ud69te"
  >
    <header
      class="css-1dviuwh"
    >
      <div
        class="logo"
      >
        <img
          alt="Customer Logo"
          src="/customer-logo.svg"
        />
        <span>
          Customer Logo
        </span>
      </div>
      <div
        class="user"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-r1updj-MuiAvatar-root"
        >
          NS
        </div>
        <span
          class="name"
        >
          Name Surname
        </span>
        <button
          aria-label="Logout"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1gp9nik-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
            data-testid="LogoutRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 5h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h6c.55 0 1-.45 1-1s-.45-1-1-1H5V5z"
            />
            <path
              d="m20.65 11.65-2.79-2.79c-.32-.32-.86-.1-.86.35V11h-7c-.55 0-1 .45-1 1s.45 1 1 1h7v1.79c0 .45.54.67.85.35l2.79-2.79c.2-.19.2-.51.01-.7z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main />
    <footer
      class="css-ik1bx0"
    >
      <nav
        class="navigation"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              Contact Us
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              Terms of Service
            </a>
          </li>
        </ul>
      </nav>
    </footer>
  </div>
</div>
`;
